server/src/main/java/org/opensearch/http/AbstractHttpServerTransport.java

```diff
@@ -352,19 +362,71 @@ protected void serverAcceptedChannel(HttpChannel httpChannel) {
      * @param httpChannel that received the http request
      */
     public void incomingRequest(final HttpRequest httpRequest, final HttpChannel httpChannel) {
-        handleIncomingRequest(httpRequest, httpChannel, httpRequest.getInboundException());
+        try (ThreadContext.StoredContext ignore = threadPool.getThreadContext().stashContext()) {
+            // TODO: Add support for parsing the otel incoming tracer.
+            final SpanScope httpRequestSpanScope = tracer.startSpan(createSpanName(httpRequest), buildSpanAttributes(httpRequest));
+            HttpChannel traceableHttpChannel = new TraceableHttpChannel(httpChannel, httpRequestSpanScope);
+            handleIncomingRequest(httpRequest, traceableHttpChannel, httpRequest.getInboundException());
+        }
+    }
+
+    private String createSpanName(HttpRequest httpRequest) {
```

Comment:
The concern I have at this stage (and it will get only worst with time) is that the tracing is supposed to be "invisible" for application developers, like logging in a sense. But the codebase is convoluted with tracing related changes more and more, we should fix that right away.

How the instrumentation should look I think is something like that (this is just general idea, please feel free to challenge or improve):

```java
       final SpanScope httpRequestSpanScope = tracer.startSpan(SpanBuilder.from(httpRequest));
       HttpChannel traceableHttpChannel = new TraceableHttpChannel(httpChannel, httpRequestSpanScope);
       handleIncomingRequest(httpRequest, traceableHttpChannel, httpRequest.getInboundException());
```

The goal - none of the tracing-related internals are the application concerns anymore, the bare minimum the developer should do - provide the context (`SpanBuilder.from(httpRequest)`). The `SpanBuilder` (our class) could have predefined methods for the contexts we support (like this one, rest request, transport action, ...). We would be adding more to it but the application classes would be cleaned up.

Reply:
Totally Agree, even I was thinking that this is polluting the code a lot and slight change/bug may even break the application. I was thinking of some util but SpanBuilder is a nice idea. Will implement this and raise the revision. Thanks!

Reply:
> @reta, Shall we build Context from the SpanBuilder.from(httpRequest) method call and Context would be having the spanName, attributes. 

Thanks @Gaganjuneja , why do we need two step process here? 

```
Context context = SpanBuilder.from(httpRequest);
final SpanScope httpRequestSpanScope = tracer.startSpan(context);
```

What is the purpose of intermediary context here if it it is fed to `startSpan` and dies?

Reply:
This is just for representation it would actually be 

```
final SpanScope httpRequestSpanScope = tracer.startSpan(SpanBuilder.from(httpRequest));
```

Reply:
@reta, should we make it more simple like this

```
final SpanScope httpRequestSpanScope = SpanBuilder.from(tracer, httpRequest));
```

Reply:
I would keep only  `tracer` as the entry point for span creation

Reply:
Yes, I also realized the same and delete the comment :)